### PR TITLE
refs #3020 introduce exceptions arg alterr and alterr_diag to hold de…

### PR DESCRIPTION
**Pull request by pvanek**
…tailed error code and description as it is described in:

https://www.postgresql.org/docs/current/static/errcodes-appendix.html

Example:

```
  err : "DBI:PGSQL:TRANSACTION-ERROR"
  desc : "connection to PostgreSQL database server lost while in a transaction; transaction has been lost"
  arg : hash: (2 members)
    alterr : "57P01"
    alterr_diag : "terminating connection due to administrator command"
```

### CODE REVIEW

**Review by davidnich — APPROVED** [review]
looks good - please add a relnote

**Comment by pvanek** [comment]
ok, updated with docs/relnotes and test

File: src/QorePGConnection.cpp [hunk 1]
```diff
@@ -1735,8 +1735,14 @@ int QorePgsqlStatement::execIntern(const char* sql, ExceptionSink* xsink) {
       if (cs == CONNECTION_BAD) {
          // first check if a transaction was in progress
          bool in_trans = conn->wasInTransaction();
-         if (in_trans)
-            xsink->raiseException("DBI:PGSQL:TRANSACTION-ERROR", "connection to PostgreSQL database server lost while in a transaction; transaction has been lost");
+         if (in_trans) {
+            char* sql_state = PQresultErrorField(res, PG_DIAG_SQLSTATE);
```
Comment (sejvlond) [inline src/QorePGConnection.cpp]: const char * + message from https://www.postgresql.org/docs/9.0/static/libpq-exec.html

also below

**Review by sejvlond — CHANGES_REQUESTED** [review]

File: src/QorePGConnection.cpp [hunk 2]
```diff
@@ -1905,6 +1901,24 @@ int QorePGConnection::get_server_version() const {
 #endif
 }
 
+const QoreHashNode* QorePGConnection::getExceptionArg(PGresult *res, ExceptionSink *xsink) {
```
Comment (sejvlond) [inline src/QorePGConnection.cpp]: `const PGresult *res`

return `QoreHashNode*`

**Review by sejvlond — CHANGES_REQUESTED** [review]

**Pull request merged**